test_fuse_argmin unexpected IR in lowering

## Reporting a bug
- [x] I am using the ~~latest version~~ 0.40.0dev  version of Numba (most recent is visible in
 the change log (https://github.com/numba/numba/blob/master/CHANGE_LOG).
- [ ] I have included below a minimal working reproducer (if you are unsure how
 to write one see http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports).

For this PR https://github.com/numba/numba/pull/3270 Travis CI testing failed for combination `PYTHON=3.5 NUMPY=1.10`, looks like there's both openblas and MKL in there... but the actual error reported is Parfors related, traceback:
```
ERROR: test_fuse_argmin (numba.tests.test_parfors.TestParfors)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/numba/numba/numba/errors.py", line 601, in new_error_context
    yield
  File "/home/travis/build/numba/numba/numba/lowering.py", line 254, in lower_block
    self.lower_inst(inst)
  File "/home/travis/build/numba/numba/numba/lowering.py", line 403, in lower_inst
    func(self, inst)
  File "/home/travis/build/numba/numba/numba/npyufunc/parfor.py", line 276, in _lower_parfor_parallel
    parfor.races)
  File "/home/travis/build/numba/numba/numba/npyufunc/parfor.py", line 1206, in call_parallel_gufunc
    ary = context.make_array(aty)(context, builder, arg)
  File "/home/travis/build/numba/numba/numba/cgutils.py", line 117, in __init__
    % (outer_ref.type.pointee, value.type))
AssertionError: bad value type: expected {i8*, i8*, i64, i64, {i64, double}*, [1 x i64], [1 x i64]}, got {i8*, i8*, i64, i64, double*, [1 x i64], [1 x i64]}
```

--- a/numba/npyufunc/parfor.py
+++ b/numba/npyufunc/parfor.py
@@ -1161,7 +1161,7 @@ def call_parallel_gufunc(lowerer, cres, gu_signature, outer_sig, expr_args, expr
                     types.intp, debug_flag)])
 
     # Get the LLVM vars for the Numba IR reduction array vars.
-    redarrs = [lowerer.loadvar(x.name) for x in list(redarrdict.values())]
+    redarrs = [lowerer.loadvar(redarrdict[x].name) for x in redvars]
 
     nredvars = len(redvars)
     ninouts = len(expr_args) - nredvars

